diff --git a/configure.ac b/configure.ac
index 554c87e..1a925bc 100644
--- a/configure.ac
+++ b/configure.ac
@@ -52,6 +52,13 @@ if test -d $KERNEL_BUILD_DIR -a -d $KERNEL_SOURCE_DIR; then
 		if test x$KMAJOR = x3; then
 			KVERSION=`echo $vcl |$AWK '{printf ("%d.%d",$3 / 65536, (($3/256)%256))}'`
 		fi
+        elif test -f $KERNEL_BUILD_DIR/include/generated/uapi/linux/version.h; then
+		vcl=`$GREP LINUX_VERSION_CODE $KERNEL_BUILD_DIR/include/generated/uapi/linux/version.h`
+		KVERSION=`echo $vcl |$AWK '{printf ("%d.%d.%d",$3 / 65536, (($3/256)%256),$3 % 256)}'`
+		KMAJOR=`echo $vcl |$AWK '{printf ("%d",$3 / 65536)}'`
+		if test x$KMAJOR = x3; then
+			KVERSION=`echo $vcl |$AWK '{printf ("%d.%d",$3 / 65536, (($3/256)%256))}'`
+		fi
 	else
 		AC_MSG_ERROR([cannot find $KERNEL_BUILD_DIR/include/linux/version.h])
 	fi
